### PR TITLE
Overhaul docs toolchain

### LINKED ISSUE #3656: Overhaul docs build chain
### References

- as suggested on https://github.com/jupyter-widgets/ipywidgets/pull/3653#issuecomment-1362939433

### Problem

The docs don't currently build.

### Suggested Improvement

- use `myst-nb`
  - under more active development than `nbsphinx`, and provides some nice features
  - some of its typographical features translate nicely into the lab experience with `jupyterlab-myst`
  - has better notebook execution caching for iteration
  - has better observability of failures and options for executing notebooks
- use `pydata-sphinx-theme`
  - better use of screen real estate
  - slightly more complex to get a good nav structure
- use `jupyterlite` directly vs `jupyterlite-sphinx`
  - allows for `sphinx 5` (`jupyterlite-sphinx` pins to `<5`, don't know or care why)
  - would allow building the lite demo _without_ the full docs site
    - which takes a _long_ time, though that would be helped above
- add `typedoc`
  - i've been using a markdown theme, and then having `myst-nb` take care of it
    - this allows typescript APIs to be discoverable in the unified sphinx search
      - even though the output is somewhat suboptimal

### PR DESCRIPTION
**Pull request by bollwyvl**
## References

- fixes #3656

## Changes
- [x] replace `nbsphinx` with `myst-nb`
- [x] replace `sphinx-rtd-theme` with `pydata-sphinx-theme` 
- [x] use `jupyterlite` directly
- [x] update `prettier`
  - [x] apply to `.md`, `.ya?ml`
- [x] move `index.rst` to `.md`
- [x] update `lerna`
- [x] update `eslint`
- [x] update `typescript`
- [x] update CI action versions   
- [x] add `sphinx-autodoc-typehints` 
- [x] add `intersphinx`
- [x] add `sphinx-copy-button`  
- [x] add `typedoc`
- [x] add `jupyterlab-myst` 
- [x] reorganize docs structure around [4 kinds of docs](https://www.writethedocs.org/videos/eu/2017/the-four-kinds-of-documentation-and-why-you-need-to-understand-what-they-are-daniele-procida/)
- [x] clean up broken markdown links
- [x] fix galata tests

## Future Work
- [ ] add dark theme variables
- [ ] add docs for integrating with `myst-nb` (and potentially `pydata-sphinx-theme` and downstreams)
- [ ] hoist preferred CDN URL (ick) to API somewhere
- [ ] probably ship a known-good requirejs somewhere
- [ ] add anchor-level `pytest-check-links` (especially for internal URLs)
- [ ] squash more sphinx warnings, ideally getting to "fail on warning"
- [ ] update prettier to consistently apply `printWidth: 88` and `proseWrap: always`

### CODE REVIEW

**Comment by bollwyvl** [comment]
This is ready for a look on the RTD preview.

The galata stuff is broken, but at least installs... wouldn't see that as breaking.

**Comment by bollwyvl** [comment]
Additionally, this _could_ be teased into a series of PRs, but given there's _no_ docs building right now, it would take a while to figure it out piecemeal.

**Comment by maartenbreddels** [comment]
Thanks a lot for this Nick!

**Comment by bollwyvl** [comment]
Thanks all.

File: docs/lite/jupyter_lite_config.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "LiteBuildConfig": {
+    "extra_ignore_contents": [
+      "Layout Example\\.ipynb",
+      "Layout Templates\\.ipynb",
+      "Variable Inspector\\.ipynb"
+    ],
+    "contents": ["../source/examples"],
+    "piplite_urls": [
```
Comment (jtpio) [inline docs/lite/jupyter_lite_config.json:9]: Shouldn't this be moved under `PipliteAddon` now with the latest lite releases?

**Comment by bollwyvl** [comment]
Good catch!

**Pull request merged**